e2e_tests/tests/command/test_shell.py

```diff
@@ -43,3 +43,51 @@ def test_open_shell() -> None:
         child.read()
         child.wait()
         assert child.exitstatus == 0
+
+
+@pytest.mark.e2e_cpu
+def test_user_flag_shell() -> None:
+    with cmd.interactive_command("shell", "start", "--detach") as shell:
+        task_id = shell.task_id
+        assert task_id is not None
+
+        child = det_spawn(["shell", "open", task_id])
+        child.setecho(True)
+        child.expect(r".*Permanently added.+([0-9a-f-]{36}).+known hosts\.", timeout=180)
+        child.sendline("det -u admin user logout")
```

Comment:
nit: instead of directly using admin, you might consider using importing from `ADMIN_CREDENTIALS` to get the user name and password, in the event that the admin credentials for testing ever change

```
from .cluster.test_users import ADMIN_CREDENTIALS
...
a_username, a_password = ADMIN_CREDENTIALS
```